### ISSUE TITLE
New 3.8.9M beta

### ISSUE THREAD

**Issue by procount** [procount/pinn#807]
There's a new beta version about to drop 😃 
I'd appreciate some testing to find any bugs.

This version builds in the ability to change the partition sizes like on Matt's pinn.mjh.nz webpage for new installations.

There's no documentation yet, so just use it to install some OSes on a new SD card and see if you can work it out! Hopefully it's fairly intuitive.

It's still in beta, so expect bugs and problems.
The Pi5 version is still prone to locking up, for example, Only the GUI locks up, but it is necessary to pull the power as the mouse/keyboard is not working. If you have ssh access, you can still login and issue a rebootp command to recover.
Any questions, just ask.

**Comment by JanSykor** [comment]
Hello @procount 
When I testing your new version on Pi3 I have problem with OSMC installation - there was some error messages with missing some noobs configs files and partition size tool not started - probably related to ability change of partition size - when I unselect OSMC and select Project space instead it was working.
Missing files was: os-pi2.json, partition_setup.sh, marketing.tar, partitions-pi2.json and NOOBS-Logo-2.png from http://download.osmc.ctv/installers/noobs/
I think it is not related, but on previous installation (not sure, maybe 3.8.9f?) on Pi5 I have back old Pinn background which I like it and in this version I have "cube" background.
In the tool "Adjust size of OSes" is possible to adjust only Extra (MB)? It could be possible to adjust also Nominal (MB) - it is gray now?
I experienced in the past, that some systems is not possible to install in Project space because low memory available in nominal section.

Thanks 
Jan

**Comment by procount** [comment]
It looks like OSMC has disappeared. I will check with the maintainer. (EDIT: Oh no, it is still there. Need to do some additional checks)
I messed up the background, sorry - it will be fixed in next version.
You can only edit the Extra field. Nominal is the minimum space required. You cannot change this. I haven't figured out a way to stop this field being selected, although you cannot edit it.
ProjectSpaces are just a generic partition layout. They are only suitable place holders for OSes that require the standard Boot/Root partition layout. If you want them for other layouts, you can create your own. Having said that, they were created a long time ago when OSes were small, so maybe it is time to increase the size of their boot partition to 512MB.
Another version about to drop soon.....

**Comment by procount** [comment]
Maybe it was just a temporary internet glitch. I can download all of the OSMC versions without problem.

**Comment by procount** [comment]
Project space partitions are now 512MB each

**Comment by JanSykor** [comment]
> Maybe it was just a temporary internet glitch. I can download all of the OSMC versions without problem.

I can confirm, that OSMC installation and bigger Project space is now working.

Reagarding Nominal size it was misunderstanding - I thought that nominal means boot partition, not minimal total memory.
Now i understand what you meant.

Thanks 
Jan

**Comment by procount** [comment]
3.8.9o beta is now available for update

**Comment by JanSykor** [comment]
After update to 3.8.9o version I still have not old Pinn wallpaper background. It was already changed?
Thanks 
Jan

**Comment by procount** [comment]
How big is your RECOVERY partition? 63MB or 127MB?

**Comment by JanSykor** [comment]
I have 128 MB
There is 2 jpg file wallpaper and wallpaper2 - but both is same picture and size
![image](https://github.com/procount/pinn/assets/23189392/04fafe41-5015-47ee-8343-1bcd5ffdca19)

Edit: When I am looking at file date, looks like it was not updated during todays update, but I dont know why, update finished OK

**Comment by procount** [comment]
On your PINN recovery partition, could you please try editing BUILD-DATA?
On the 2nd line, change 3.8.9o to 3.8.9 by just removing the 'o'.
This will force PINN to update itself again on next boot.

I have just updated the zip file, so provided the mirrors have updated, it should be corrected on next boot.
If it doesn't fix it, please wait a few minutes for the mirrors to update and repeat.

**Comment by JanSykor** [comment]
Yes it helped
Now I have correct backgroud
Thank you so much. 
Jan

**Comment by procount** [comment]
I'm hoping this is the last beta before merging the Pi5 into mainstream, so please give it a good testing, especially the partition resizing as it is new.
The Pi5 still has the same occasional gui lockup, no CEC, no DSI display, no Hyperpixel, no additional wifi etc, but I will try to address those after the merge.

**Comment by andriilive** [comment]
@procount noticed joystic drivers, cool :D  should not **vl805** be included here also?

Raspberry Pi 4 EEPROM bootloader rescue image
*********************************************

The Raspberry Pi 4 contains a small EEPROM used to store the bootloader.

This rescue image reverts the bootloader EEPROM to factory default settings.

This rescue image also updates the USB 3 controller (VL805) firmware to the
latest version, 138a1, which has better full-speed isochronous endpoint 
support.

Raspberry Pi 4 board revisions 1.1 and 1.2 contain a separate EEPROM 
which contains firmware for the USB 3 controller (VL805): on newer revisions 
the USB controller firmware is stored in the bootloader EEPROM along with 
the bootloader.

![CleanShot 2024-10-08 at 03 26 26@2x](https://github.com/user-attachments/assets/50c7acbd-86a3-414b-a7ee-ff89e9193cd4)

**Comment by procount** [comment]
Updating of any vl805 or other EEPROMS is best done with the already supplied RPF tools. I'm not sure I need to duplicate this.

**Comment by andriilive** [comment]
> Updating of any vl805 or other EEPROMS is best done with the already supplied RPF tools. I'm not sure I need to duplicate this.

I’m new to hardware, but this involves a USB 3.0 drive. My external drive turns off during each boot to PINN after the splash screen appears, even when it’s connected to an external power source via the cable, 

![image](https://github.com/user-attachments/assets/913c8ccf-44e9-4653-97a9-5a39f78681fe)
![image](https://github.com/user-attachments/assets/a56296e0-ed63-427f-9622-bb898c778de4)

I guess a driver is a good start to start investigation, so as far I understand:

1. I tried booting with the SD card and drive inserted. A day before yesterday the drive was disconnecting twice and boot stucked at end, only booting without SD card worked. Yesterday  i've updated the boot to new nvm -> usb -> sd version + pi imager via the brew. Now it sticks only once and boots successfully, if i plug off and plug in the drive leaving sd card on its place

3. Drivers should be part of the boot process, as far i understand? (I notice a correlation between PINN’s boot sector and the project’s boot. There might be bash scripts involved, but the version from Pi Imager is messy. Also, there’s **no clear timeline for hooks** in the documentation)

**Comment by andriilive** [comment]
Here are some minor bugs I found, most of them are related to pi-imager settings

![image](https://github.com/user-attachments/assets/1aa67730-441b-4f9f-9a0d-19e5edfd703a)

https://www.raspberrypi.com/documentation/computers/getting-started.html#advanced-options

• The pi-imager generates `user-data` following its global pattern, but the `password` is probably random (or unknown). I can’t log in with root / raspberry or any of the expected passwords I’ve used in installations. The password field is disabled while the SSH key persists in the config. I'm not sure if this part is on you or on a raspberry team

![CleanShot 2024-10-11 at 09 13 41@2x](https://github.com/user-attachments/assets/52dcdee2-5417-46a3-bdab-965e1c544b29)

- the keyboard layout rewritten: (**"us"** in my case) is being ignored, **"gb"** instead is being persisted and populated across installed instances, **making me really angry on £** (i'm getting pounds instead of # comments)

- a core network config is being ignored, not a bag but can be parsed / combined with network / dhcp configs (from your docs) near on a screenshot is a config generated by the imager

![CleanShot 2024-10-11 at 09 21 10@2x](https://github.com/user-attachments/assets/65eb7fc5-81b3-4a3f-9175-b05f5d7cebda)

- same as above mentioned, dozens of errors when I tried "**download**" second time with PINN, just downloaded manually to sd card and installed from there. I guess the reason of error was distros mismatch because of the mess in tools

I flushed an image with **'pi-imager'** did a wrong partition, formatted, tried https://pinn.mjh.nz, got errors formatted again, checked a readme then flushed **pinn-light**, and so on. that's was insane 😕
**You're mentioning os partitions in explanation how Projects work, in the recovery app and docs but not explaining where this info comes from**, as far i see the only way to see them from the PINN app is download, then cat 

It was much quicker for me to download the os instances, explore the partitions.json of each and only then proceed to install. 

```bash
#!/bin/bash

# example: curl "https://sourceforge.net/projects/pinn/rss?path=/os/kali64" | grep "<link>.*</link>" | sed 's|<link>||;s|</link>||' | while read url; do url=`echo $url | sed 's|/download$||'`; wget $url ; done

# Array of OS names
array_of_os=("kali64" "raspios_armhf_full")

# Loop through each OS in the array
for os in "${array_of_os[@]}"; do
    # Create a directory for the OS and navigate into it
    mkdir "$os" && cd "$os" || exit

    # Fetch RSS feed, extract download links, and download the files
    curl "https://sourceforge.net/projects/pinn/rss?path=/os/$os" \
    | grep "<link>.*</link>" \
    | sed 's|<link>||;s|</link>||' \
    | while read url; do
        # Clean up the URL (removes /download part)
        url=$(echo "$url" | sed 's|/download$||')
        # Download the file
        wget "$url"
    done

    # Go back to the parent directory
    cd ..
done
```

- you're doing a great job and support but the docs is a mess, while the noobs like me will continue coming **from the pi-imager to your GitHub (not otherwise)** some parts are seems bootstrapped from a previous (NOOBS) version, the other is pinn-lite, then pi-imager plus the web one https://pinn.mjh.nz. neither of them correspond, this leads to a really huge mess in configs and understanding what actually happens and when. Check this out: 

![CleanShot 2024-10-11 at 09 57 28@2x](https://github.com/user-attachments/assets/d19176ba-a85a-4393-b1da-9cb651bb7795)

It would be nice to have a view of phases to see what and when happens, like

- recovery
- config
- config.txt in each instannce generated -< config

**Comment by procount** [comment]
I'm really sorry, but I'm not understanding your issues, or how any of these relate to PINN. Maybe something is getting lost in translation...?

> My external drive turns off each boot when screen splashes, even being connected to the external power source via the cable

So I think here you are commenting that your external drive powers down temporarily after you have booted PINN  and just before you boot into one of your installed OSes...? This is normal for any drive as the Raspberry Pi will reboot to boot into one of its installed OSes. No driver will help avoid this.

The rest seems maybe to do with Android? If you have problems with the OSes you need to contact the OS maintainers. I only convert the format of the OS so that it can multiboot.

**Comment by andriilive** [comment]
Sorry for the earlier message, I was drafting on the go. Hopefully,  its is clearer after update. Just sharing a few bugs from pi-imager version 3.9.2, gaps in the manuals and a mess the noobs can face 

> You mentioned that your external drive powers down temporarily after booting into PINN.

No, it was powering down when the colored video signal was sent, before entering recovery mode. The issue disappeared later without any changes in PINN settings. 

It seems the TV was turning the Pi off and on using CEC. I had previously disabled CEC on a TV, but I hadn’t hard-rebooted it to apply the changes :/ so continued investigating false trail which brought me again to USB ports check 😑
